### PR TITLE
Add FederatedTypeConfig e2e validation webhook test

### PR DESCRIPTION
**Pull request by font**
This is a partial fix for #944.

### CODE REVIEW

File: test/e2e/validation.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+			f.Logger().Fatalf("Expected error creating invalid %s = %+v", resourceName, invalidFtc)
+		}
+
+		By(fmt.Sprintf("Creating a valid %s", resourceName))
+		validFtcCopy := validFtc.DeepCopyObject().(*v1beta1.FederatedTypeConfig)
+		err = client.Create(context.TODO(), validFtcCopy)
+		if err != nil {
+			f.Logger().Fatalf("Unexpected error creating valid %s = %+v, err: %v", resourceName, validFtcCopy, err)
+		} else {
+			framework.AddCleanupAction(func() {
```
[the remark this answers is not in the document]
Comment (font) [reply test/e2e/validation.go]: This is necessary when testing against a namespace scoped deployment i.e. when the test namespace is the kubefed system namespace, as the containing namespace won't be automatically removed.

**Comment by xunpan** [comment]
LGTM

**Comment by marun** [comment]
@font Should this PR be held pending the outcome of your discussion with David?

**Comment by font** [comment]
@marun I don't think we should hold it as more will need to change beyond this e2e test.

File: test/e2e/validation.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (marun) [reply test/e2e/validation.go]: Why would it be desirable to run the test against a namespace-scoped deployment vs skipping?  That implies creating an FTC in a namespace monitored by the ftc controller.

[the remark this answers is not in the document]
Comment (font) [reply test/e2e/validation.go]: I think testing that we properly fail validation of an invalid FTC when the FTC is created in a namespace monitored by the FTC controller i.e. the namespace-scoped deployment case, is vital. You don't agree?

**Comment by marun** [comment]
LGTM

**Pull request merged**